moirae/extractors/ecm.py

```diff
@@ -210,15 +206,18 @@ def interpolate_r0(self, cycle: pd.DataFrame) -> np.ndarray:
         """
         # calculate soc throughout cycle
         cycle = cycle.copy(deep=False)
-        cycle = self.soc_calc(cycle)
+        if 'cycle_capacity' not in cycle.columns:
+            StateOfCharge().enhance(cycle)
+        cycle['soc'] = (cycle['cycle_capacity'] - cycle['cycle_capacity'].min()) / \
+                       (cycle['cycle_capacity'].max() - cycle['cycle_capacity'].min())
 
         # calculate instantanous resistance at all points
         cycle['r0_inst'] = np.abs(
             cycle['voltage'].diff() /
             cycle['current'].diff())
 
         # calculate key quantities to filter R0_inst
-        Inorm = cycle['capacity'].max()
+        Inorm = cycle['cycle_capacity'].max() - cycle['cycle_capacity']

```

Comment:
I don't think this is the correct thing to do. We should keep the original as we want it to be a scalar quantity that we use to normalize the change in current.